concent_api/core/models.py

```diff
@@ -73,3 +77,236 @@ class MessageAuth(Model):
+        ):
+            raise ValidationError({
+                'state': 'Subtask cannot change its state from {} to {}.'.format(
+                    self._current_state, self.state
+                )
+            })
+        else:
+            self._current_state = self.state
+
+
+class PaymentCommittedMessages(models.Model):
+    """
+    Stores messages with payments for requestor to be returned from the 'receive-out-of-band' endpoint
+    """
+
+    class RecipientType(ChoiceEnum):
+        Provider    = 'provider'
+        Requestor   = 'requestor'
+
+    payment_ts              = models.DateTimeField()
+    task_owner_key          = Base64Field(max_length = 64)
+    provider_eth_account    = Base64Field(max_length = 64)
+    amount_paid             = models.FloatField()
+    recipient_type          = models.CharField(max_length = 32, choices = RecipientType.choices())
+    amount_pending          = models.FloatField()
+
+
+class PendingResponse(models.Model):
+    """
+    Stores information about messages to be returned from the `receive` or `receive-out-of-band` endpoint.
+    """
+
+    class ResponseType(ChoiceEnum):
+        ForceReportComputedTask         = 'ForceReportComputedTask'
+        ForceSubtaskResultsSettled      = 'ForceSubtaskResultsSettled'
+        AckReportComputedTask           = 'AckReportComputedTask'
+        RejectReportComputedTask        = 'RejectReportComputedTask'
+        VerdictReportComputedTask       = 'VerdictReportComputedTask'
+        ForceGetTaskResult              = 'ForceGetTaskResult'
+        ForceGetTaskResultRejected      = 'ForceGetTaskResultRejected'
+        ForceGetTaskResultFailed        = 'ForceGetTaskResultFailed'
+        ForceGetTaskResultUpload        = 'ForceGetTaskResultUpload'
+        ForgetGetTaskResultDownload     = 'ForgetGetTaskResultDownload'
+        ForceSubtaskResults             = 'ForceSubtaskResults'
+        SubtaskResultsSettled           = 'SubtaskResultsSettled'
+        SubtaskResultsAccepted          = 'SubtaskResultsAccepted'
+        SubtaskResultsRejected          = 'SubtaskResultsRejected'
+        ForcePaymentCommitted           = 'ForcePaymentCommitted'
+
+    class PendingResponseType(ChoiceEnum):
+        Receive             = 'receive'
+        ReceiveOutOfBand    = 'receive_out_of_band'
+
+    response_type        = models.CharField(max_length = 32, choices = ResponseType.choices())
+    client               = models.ForeignKey(Client)
+    queue                = models.CharField(max_length = 32, choices = PendingResponseType.choices())
+
+    # TRUE if the client has already fetched the message.
+    delivered            = models.BooleanField(default = False)
+
+    subtask              = models.ForeignKey(Subtask, blank = True, null = True)
+    created_at           = models.DateTimeField(default = timezone.now)
+    payment_message      = models.OneToOneField(PaymentCommittedMessages, blank = True, null = True)
```

Comment:
According to @cameel comment, this relation should be on `PaymentCommittedMessages` model, off course reversed. You can use `rel_name` attribute to set the attribute name on which `PendingResponse` can use `PaymentCommittedMessages` to avoid changes in views.